src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContextBuilder.cs

```diff
@@ -111,6 +111,22 @@ public virtual MsBuildProjectContext Build(bool ignoreBuildErrors = false)
             return new MsBuildProjectContext(name, _configuration, projectInstance);
         }
 
+        public IEnumerable<MsBuildProjectContext> BuildAllConfigurations()
```

Comment:
I'd change this to `MsBuildProjectContext Build(NuGetFramework framework)`.

No need to save the caller a for loop if not needed.